### PR TITLE
Add support for using reverse tunnel for rsyslog

### PR DESCRIPTION
**Pull request by fruch**
## TODOs:
- [x] Test with scylla-cloud
- [x] Test with multi-region

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (`hydra unit-tests`)
- ~~[ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/auto_ssh.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    local_runner = LocalCmdRunner()
+    local_runner.run("docker kill {docker_name}-{host_name}-autossh".format(**locals()), ignore_status=True)
+
+
+if __name__ == "__main__":
+
+    class DummyNode(object):
+        _ssh_login_info = {'hostname': '34.244.186.231',
+                           'user': 'centos',
+                           'key_file': '/home/fruch/.ssh/scylla-qa-ec2',
```
Comment (bentsi) [inline sdcm/auto_ssh.py]: create a proper unit test

Comment (bentsi) [reply sdcm/auto_ssh.py]: or remove this code

File: sdcm/cluster.py [hunk 2]
```diff
@@ -390,6 +397,10 @@ def __init__(self, name, parent_cluster, ssh_login_info=None, base_logdir=None,
         self._cassandra_stress_version = None
         self.lock = threading.Lock()
 
+        if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION:
+            from sdcm.auto_ssh import start_auto_ssh
```
Comment (bentsi) [inline sdcm/cluster.py]: why did you put the import here?

File: sdcm/cluster.py [hunk 3]
```diff
@@ -270,10 +270,17 @@ def configure_rsyslog(cls, enable_ngrok=False):
...
         if cls.RSYSLOG_ADDRESS:
-            post_boot_script += dedent('''
-                   sudo echo 'action(type="omfwd" Target="{0}" Port="{1}" Protocol="tcp")'>> /etc/rsyslog.conf
-                   sudo systemctl restart rsyslog
-                   '''.format(*cls.RSYSLOG_ADDRESS))
+            if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION:
+                post_boot_script += dedent('''
+                       sudo echo 'action(type="omfwd" Target="{0}" Port="{1}" Protocol="tcp")'>> /etc/rsyslog.conf
+                       sudo systemctl restart rsyslog
+                       '''.format('127.0.0.1', '5000'))
```
Comment (bentsi) [inline sdcm/cluster.py]: create `SSH_TUNNEL_PORT` global in the `auto_ssh.py`

File: sdcm/cluster.py [hunk 4]
```diff
@@ -390,6 +397,10 @@ def __init__(self, name, parent_cluster, ssh_login_info=None, base_logdir=None,
         self._cassandra_stress_version = None
         self.lock = threading.Lock()
 
+        if IP_SSH_CONNECTIONS == 'public' or Setup.MULTI_REGION:
+            from sdcm.auto_ssh import start_auto_ssh
+            start_auto_ssh(Setup.test_id(), self, Setup.RSYSLOG_ADDRESS[1], '5000')
```
Comment (bentsi) [inline sdcm/cluster.py]: use `SSH_TUNNEL_PORT` from the comment above as a default in the `start_auto_ssh` func

File: sdcm/auto_ssh.py (hunk 1 above)
Comment (fruch) [reply sdcm/auto_ssh.py]: I'll go with removing

File: sdcm/cluster.py (hunk 3 above)
Comment (fruch) [reply sdcm/cluster.py]: Add a global for it

File: sdcm/cluster.py (hunk 2 above)
Comment (fruch) [reply sdcm/cluster.py]: forgot to move it into place

Comment (fruch) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py (hunk 4 above)
Comment (fruch) [reply sdcm/cluster.py]: done

File: sdcm/cluster_baremetal.py [hunk 5]
```diff
@@ -60,6 +63,22 @@ def restart(self):
     def destroy(self):
         self.stop_task_threads()  # For future implementation of destroy
 
+    def run_startup_script(self):
+        tmp_file = tempfile.NamedTemporaryFile(delete=False)
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_baremetal.py]: done

File: sdcm/cluster_baremetal.py [hunk 6]
```diff
@@ -60,6 +63,22 @@ def restart(self):
...
+        tmp_file.write(cluster.Setup.get_startup_script())
+        tmp_file.close()
+
+        self.remoter.send_files(src=tmp_file.name, dst='/tmp/sct-startup.sh')
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster_baremetal.py]: done

**Pull request merged**